src/vs/workbench/services/textfile/browser/textFileService.ts

```diff
@@ -108,27 +110,29 @@ export abstract class TextFileService implements ITextFileService {
 		this.toUnbind.push(this.editorGroupService.onEditorsChanged(() => this.onEditorFocusChanged()));
 	}
 
-	private beforeShutdown(): boolean | TPromise<boolean> {
+	private beforeShutdown(quitRequested: boolean): boolean | TPromise<boolean> {
```

Comment:
A few reasons:

- `TextFileService` depends on `BackupService.cleanupBackupsBeforeShutdown` because it needs to clean up backups.
- `BackupService` needs to trigger `TextFileService.confirmBeforeShutdown`.
- Even if they were separate, there would need to be some communication between the 2 services. Depending on what `BackupService` does, it may need to not give a prompt even if there are dirty files for example.

If you have any ideas how to get around this nicely that would be great :smiley: